subsys/bluetooth/controller/Kconfig

```diff
@@ -142,63 +142,6 @@ config BT_CTLR_HCI_VS_BUILD_INFO
 	  character is required at the beginning to separate it from the
 	  already included information.
 
-config BT_CTLR_AD_DATA_BACKUP
-	bool "Legacy AD Data backup"
-	depends on BT_PERIPHERAL || BT_CTLR_ADV_EXT
-	default y
-	help
-	  Backup Legacy Advertising Data when switching to Legacy Directed or
-	  to Extended Advertising mode, and restore it when switching back to
-	  Legacy Non-Directed Advertising mode.
-	  Application can disable this feature if not using Directed
-	  Advertising or switch between Legacy and Extended Advertising.
-
-config BT_CTLR_HCI_ADV_HANDLE_MAPPING
-	bool "Advertising set handle mapping between HCI and LL"
-	depends on BT_CTLR_ADV_EXT
-	default y if BT_HCI_RAW
-	help
-	  Enable mapping of advertising set handles between HCI and LL when
-	  using external host since it can use arbitrary numbers as set handles
-	  (as defined by Core specification) as opposed to LL which always uses
-	  zero-based numbering. When using with Zephyr host this option can be
-	  disabled to remove extra mapping logic.
-
-config BT_CTLR_DUP_FILTER_LEN
-	int "Number of addresses in the scan duplicate filter"
-	depends on BT_OBSERVER
-	default 16
-	help
-	  Set the number of unique BLE addresses that can be filtered as
-	  duplicates while scanning.
-
-config BT_CTLR_DUP_FILTER_ADV_SET_MAX
-	int "Number of Extended Advertising Sets in the scan duplicate filter"
-	depends on BT_OBSERVER && BT_CTLR_ADV_EXT && (BT_CTLR_DUP_FILTER_LEN > 0)
-	range 1 16
-	default 1
-	help
-	  Set the number of unique Advertising Set ID per Bluetooth Low Energy
-	  addresses that can be filtered as duplicates while Extended Scanning.
```

Comment:
Duplicate filtering is standard Bluetooth LE Feature, hence should be configurable in the top level Kconfig.